stdlib/tests/main.rs

```diff
@@ -0,0 +1,190 @@
+mod crypto;
+mod math;
+mod mem;
+mod sys;
+
+use miden_stdlib::StdLibrary;
+use processor::{AdviceInputs, MerkleSet, StackInputs};
+use processor::{ExecutionError, ExecutionTrace, MemAdviceProvider};
+use proptest::prelude::*;
+use std::collections::BTreeMap;
+use vm_core::{stack::STACK_TOP_SIZE, Felt, FieldElement, Program};
+
+/// Following section suffers from code-duplication, why ?
+///
+/// Test helper routines were originally living in `miden-vm` crate's
+/// integration test helper module ( ../miden/tests/integration/helpers/mod.rs ).
+/// But for addressing issue described in https://github.com/0xPolygonMiden/miden-vm/issues/723
+/// we move standard library related tests out of `miden-vm` crate and put them
+/// here inside `miden-stdlib` crate. But we can't just export test helper routines
+/// from `miden-vm` crate and start using that here by simply importing them - because
+/// `miden-vm` crate is an aggregation of all components of this repository, meaning it's
+/// dependent on `miden-stdlib` ( this crate ) too. Attempting to bring in test helper
+/// routines from `miden-vm` crate results in circular dependency.
+///
+/// I think we can solve this code duplication problem by keeping test helper routines
+/// here in `miden-stdlib` crate and by making them exportable, we ensure that they can
+/// be used in `miden-vm` crate, which anyway imports this crate as development dependency.
```

Comment:
This could work - though, I think we'd also need to import the verifier here so that we can provide `prove_and_verify()` method. However, I'm not sure this is the cleanest approach.

Maybe a better way is to have a separate crate for test helpers. For example, something like `miden-test` or `miden-test-utils`. Then this crate could be used here, and `miden-vm` and in any other libraries we'll build (e.g., in `miden-base`). This crate can then also re-export common things we usually need during testing - ie.g., `rand-utils`, `proptest`, `test-case` etc.

If we do decide to go this route, building the test utils crate should probably be done in a different PR.